util.go

```diff
@@ -0,0 +1,80 @@
+package files
+
+import (
+	"bytes"
+	"io"
+	"io/ioutil"
+	"sort"
+)
+
+// ToFile is an alias for n.(File). If the file isn't a regular file, nil value
```

Comment:
Do we have use-cases for `ToFile`/`ToDir` outside of tests? If not, I'd just make these test helpers.

Otherwise, we should consider `AssertFile`, `AssertDir`. Basically, if the user isn't sure, I'd expect them to use a type-switch.

Reply:
Those are useful in places which take generic type and then do different things depending on the type, for example `addFile` in go-ipfs adder does something along the lines of:

```go
func (a *adder) addFile(n files.Node) {
[...]
if d := files.ToDir(n); d != nil {
  return a.addDir(d)
}
[large chunk of code...]
```

It could just use a type switch, but this form is shorter and arguably nicer.

Reply:
I see. I just don't want someone writing:

```go
if d := files.ToDir(n); d != nil {
  return a.addDir(d)
}
f := files.ToFile(n)
// use f without checking
```

But I can live with that.